core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -126,21 +121,23 @@ public TieredBlockStore() {
     mMetaManager = BlockMetadataManager.createBlockMetadataManager();
     mLockManager = new BlockLockManager();
 
+    mBlockIterator = mMetaManager.getBlockIterator();
+    // Register listeners required by the block iterator.
+    for (BlockStoreEventListener listener : mBlockIterator.getListeners()) {
+      registerBlockStoreEventListener(listener);
+    }
+
     BlockMetadataEvictorView initManagerView = new BlockMetadataEvictorView(mMetaManager,
         Collections.<Long>emptySet(), Collections.<Long>emptySet());
     mAllocator = Allocator.Factory.create(initManagerView);
     if (mAllocator instanceof BlockStoreEventListener) {
       registerBlockStoreEventListener((BlockStoreEventListener) mAllocator);
     }
 
-    initManagerView = new BlockMetadataEvictorView(mMetaManager, Collections.<Long>emptySet(),
-        Collections.<Long>emptySet());
-    mEvictor = Evictor.Factory.create(initManagerView, mAllocator);
-    if (mEvictor instanceof BlockStoreEventListener) {
-      registerBlockStoreEventListener((BlockStoreEventListener) mEvictor);
-    }
-
-    mStorageTierAssoc = new WorkerStorageTierAssoc();
+    // Initialize and start coordinator.
+    mTaskCoordinator = new ManagementTaskCoordinator(this, mMetaManager,
+        new DefaultStoreLoadTracker(), () -> getUpdatedView());
+    mTaskCoordinator.start();
```

Comment:
Is it okay to have this thread running by itself, should we keep track of it?

Reply:
Closing this now as part of block-worker shutdown.